src/nvim/testdir/test_signals.vim

```diff
@@ -0,0 +1,140 @@
+" Test signal handling.
+
+source check.vim
+source term_util.vim
+
+CheckUnix
+
+source shared.vim
+
+" Check whether a signal is available on this system.
+func HasSignal(signal)
+  let signals = system('kill -l')
+  return signals =~# '\<' .. a:signal .. '\>'
+endfunc
+
+" Test signal WINCH (window resize signal)
+func Test_signal_WINCH()
+  throw 'skipped: Nvim cannot avoid terminal resize'
+  if has('gui_running') || !HasSignal('WINCH')
+    return
+  endif
+
+  " We do not actually want to change the size of the terminal.
+  let old_WS = ''
+  if exists('&t_WS')
+    let old_WS = &t_WS
+    let &t_WS = ''
+  endif
+
+  let old_lines = &lines
+  let old_columns = &columns
+  let new_lines = &lines - 2
+  let new_columns = &columns - 2
+
+  exe 'set lines=' .. new_lines
+  exe 'set columns=' .. new_columns
+  call assert_equal(new_lines, &lines)
+  call assert_equal(new_columns, &columns)
+
+  " Send signal and wait for signal to be processed.
+  " 'lines' and 'columns' should have been restored
+  " after handing signal WINCH.
+  exe 'silent !kill -s WINCH ' .. getpid()
+  call WaitForAssert({-> assert_equal(old_lines, &lines)})
```

Comment:
This always fails.

Reply:
Maybe related to https://github.com/neovim/neovim/issues/11330.

Reply:
Related but different issue. This test is suppose to not resize the terminal because `t_WS` option is set to empty string. Neovim does ignore `SIGWINCH` so `lines` and `columns` are not restored to the old values.